src/interpret.ml

```diff
@@ -99,20 +99,61 @@ module Make (P : Interpret_intf.P) :
 
   let consti i = const_i32 (Int32.of_int i)
 
+  let clz_impl_32 n =
+    let rec aux (lb : int) ub =
+      if ub = lb + 1 then return (const_i32 (Int32.of_int (32 - ub)))
+      else begin
+        let mid = (lb + ub) / 2 in
+        let two_pow_mid = Int32.shl 1l (Int32.of_int mid) in
+        let> cond = I32.(lt_u n (const_i32 two_pow_mid)) in
+        if cond then aux lb mid else aux mid ub
+      end
+    in
+    let> cond = I32.(eqz n) in
+    if cond then return @@ const 32l else aux 0 32
```

Comment:
Could you explain to me quickly why this can't be in `symbolic_value` directly ?

Reply:
Because currently, code in symbolic_value has no reference to the Choice monad

Reply:
What about putting it in `symbolic.ml` then ?

Replacing:

```ocaml
  module Value = Symbolic_value
```

By:

```ocaml
  module Value = struct
    include Symbolic_value
    let clz ...
  end
```

?

Reply:
I can try but it is awkward as well because we would have to separate cases for I64 and I32 in Symbolic.value

Reply:
Oh yes I forgot about this. I still believe you can do it "properly":

Reply:
```ocaml
module I32 = struct
  include Symbolic_value.I32
  let clz = ...
end

module I64 = struct
  ... same ...
end

module Value = struct
  include Symbolic_value
  module I32 = I32
  module I64 = I64
end
```